tasks/smoke-test/playwright-fixtures/rwServe.fixture.ts

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable no-empty-pattern */
+import { test as base } from '@playwright/test'
+import execa from 'execa'
+
+import { waitForServer } from '../util'
+
+// Declare worker fixtures.
+type DevServerFixtures = {
+  port: number
+  server: any
+}
+
+// Note that we did not provide an test-scoped fixtures, so we pass {}.
+const test = base.extend<any, DevServerFixtures>({
+  port: [
+    async ({}, use, workerInfo) => {
+      // "port" fixture uses a unique value of the worker process index.
+      await use(8899 + workerInfo.workerIndex)
+    },
+    { scope: 'worker' },
+  ],
+
+  // "server" fixture starts automatically for every worker - we pass "auto" for that.
+  server: [
+    async ({ port }, use) => {
+      console.log('Starting rw server.....')
+
+      const projectPath = process.env.PROJECT_PATH
+
+      if (!projectPath) {
+        throw new Error(
+          'PROJECT_PATH not defined. Need this to launch the dev server'
+        )
+      }
+
+      console.log(`Running rw serve at ${projectPath}`)
+
+      // Don't wait for this to finish, because it doens't
+      const rwServeHandler = execa.command(`yarn rw serve -p ${port}`, {
+        cwd: projectPath,
+        shell: true,
+      })
```

Comment:
This test wasn't working for me:
```
dom@evaM1 ~/p/r/r/t/smoke-test (chore/smoke-test) [1]> PROJECT_PATH=~/prjcts/rw/_sandbox/tp npx playwright test rwServe

Running 1 test using 1 worker

  ✘  tests/rwServe.spec.ts:5:1 › Smoke test with rw serve (1m)
Starting rw server.....
Running rw serve at /Users/dom/prjcts/rw/_sandbox/tp
Waiting for dev servers.....
yarn run v1.23.0-20220130.1630
$ /Users/dom/prjcts/rw/_sandbox/tp/node_modules/.bin/rw serve -p 8899
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.


  1) tests/rwServe.spec.ts:5:1 › Smoke test with rw serve ==========================================

    Timeout of 60000ms exceeded.

  Slow test file: tests/rwServe.spec.ts (1m)
  Consider splitting slow test files to speed up parallel execution

  1 failed
    tests/rwServe.spec.ts:5:1 › Smoke test with rw serve ===========================================
```

So I went into my test project and tried running `yarn rw serve` to see if it'd work outside the test. It was a familiar gotcha for me:

```
dom@evaM1 ~/p/r/_/tp> yarn rw serve
yarn run v1.23.0-20220130.1630
$ /Users/dom/prjcts/rw/_sandbox/tp/node_modules/.bin/rw serve
(node:3152) ExperimentalWarning: stream/web is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)

 Please run `yarn rw build` before trying to serve your redwood app. 

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Looks like we just need to add:

```ts
await execa.command(`yarn rw build`, {
  cwd: projectPath,
  shell: true,
})
```

Reply:
But I know we're running build before the smoke tests in CI:

https://github.com/redwoodjs/redwood/blob/04d0e9780826f2ef488a9132046f33e840015c03/.github/workflows/smoke-test.yaml#L55-L65

Maybe we can have the serve fixture do it and remove it from the yaml file?

Reply:
I've thought about it... but I think the playwright fixtures are meant to be for shorter running tasks. 


They're also intended to be used for parallelising tests. So each worker (if we started parallelising) would spin up its own fixture... so you can imagine, generating a new project with the yarn installs isn't a viable option.

Reply:
I've added a few extra lines to handle this now. If the project is already built (i.e. it has web/dist and api/dist, it won't build, otherwise it'll build once)